From: Jan Kara <jack@suse.cz>
Subject: Increase limits for starting writeback of dirty data
References: bnc#449662
Patch-mainline: ?

Return limits for dirty pages writeback to the numbers from SLES10. This dramatically
improves performance of workloads dirtying a lot of memory (e.g. simple databases not
using direct IO) and we're not aware it would harm anything.

Signed-off-by: Jan Kara <jack@suse.cz>

diff -rupX /home/jack/.kerndiffexclude linux-2.6.27/mm/page-writeback.c linux-2.6.27-dirty_limits//mm/page-writeback.c
--- linux-2.6.27/mm/page-writeback.c	2008-10-10 00:13:53.000000000 +0200
+++ linux-2.6.27-dirty_limits//mm/page-writeback.c	2009-01-07 16:09:24.000000000 +0100
@@ -66,7 +66,7 @@ static inline long sync_writeback_pages(
 /*
  * Start background writeback (via pdflush) at this percentage
  */
-int dirty_background_ratio = 5;
+int dirty_background_ratio = 10;
 
 /*
  * free highmem will not be subtracted from the total free memory
@@ -77,7 +77,7 @@ int vm_highmem_is_dirtyable;
 /*
  * The generator of dirty data starts writeback at this percentage
  */
-int vm_dirty_ratio = 10;
+int vm_dirty_ratio = 40;
 
 /*
  * The interval between `kupdate'-style writebacks, in jiffies
